test/integration/telemetry_test.exs

```diff
@@ -131,6 +122,19 @@ defmodule Oban.Integration.TelemetryTest do
     :telemetry.detach("oban-default-logger")
   end
 
+  test "span publishes given metric and metadata" do
+    :telemetry.attach("span-handler", [:oban, :test_metric], &Handler.handle/4, self())
+
+    assert :example_return = Telemetry.span(:test_metric, fn -> :example_return end)
+    assert :ok = Telemetry.span(:test_metric, fn -> :timer.sleep(5) end, %{example_metadata: 1})
+
+    assert_receive {:executed, :test_metric, _duration, %{}}
+    assert_receive {:executed, :test_metric, duration, %{example_metadata: 1}}
+    assert duration > 5
```

Comment:
The `sleep` is for 5 milliseconds and the duration is reported in microseconds. I think asserting that this is greater than 0 is sufficient.